lib/uklock/include/uk/mutex.h

```diff
@@ -52,16 +54,21 @@
 extern "C" {
 #endif
 
+#define UK_MUTEX_CONFIG_WRITE_RECURSE 0x01
+
 /*
  * Mutex that relies on a scheduler
  * uses wait queues for threads
  */
 struct uk_mutex {
 	int lock_count;
+	uint8_t config_flags;
```

Comment:
Due to padding you do not gain any savings by using `uint8_t`. In fact, since you have to use longer `movzx` instructions to zero-extend the value during load, it can happen that the OPCODE becomes even longer. Although that depends on the code. 

Nevertheless, since you are using an `unsigned int` in `uk_rwlock`, I would use the same here for consistency.

```suggestion
	unsigned int config_flags;
```